File: update_videos.py
import json
import re

try:
    with open('src/data/videoData.js', 'r', encoding='utf-8') as f:
        content = f.read()
    
    with open('playlist.json', 'r', encoding='utf-8') as f:
        playlist = json.load(f)
    
    new_videos = []
    for v in playlist:
        ep_match = re.search(r'ep\s*(\d+)', v['title'], re.IGNORECASE)
        eps_str = '文藝茶館第一季'
        if ep_match:
            eps_str += ' EP' + ep_match.group(1)
            
        title_escaped = v['title'].replace("'", "\\'")
        
        video_str = f"""  {{
    id: '{v['id']}',
    title: '{title_escaped}',
    series: 'wenyi_s1',
    category: 'all',
    duration: '{v['duration']}',
    description: '{eps_str}'
  }},"""
        new_videos.append(video_str)
        
    new_videos_str = '\n'.join(new_videos)
    
    new_content = content.replace(
        "export const videos = [",
        "export const videos = [\n" + new_videos_str
    )
    
    if 'wenyi_s1' not in new_content:
        new_content = new_content.replace(
            "{ id: 'wenyi', label: '文藝茶館 第二季' },",
            "{ id: 'wenyi_s1', label: '文藝茶館 第一季' },\n  { id: 'wenyi', label: '文藝茶館 第二季' },"
        )
        
        new_content = new_content.replace(
            "export const WENYI_SUBCATEGORIES = [",
            "export const WENYI_S1_SUBCATEGORIES = [\n  { id: 'all', label: '全部' }\n];\n\nexport const WENYI_SUBCATEGORIES = ["
        )
        
    with open('src/data/videoData.js', 'w', encoding='utf-8') as f:
        f.write(new_content)
        
    print("Successfully updated videoData.js")

except Exception as e:
    print(f"Error: {e}")

File: update_videos.js
/* global require */
const fs = require('fs');

try {
    const content = fs.readFileSync('src/data/videoData.js', 'utf8');
    const playlist = JSON.parse(fs.readFileSync('playlist.json', 'utf8'));

    const newVideosStr = playlist.map(v => {
        const epMatch = v.title.match(/ep\s*(\d+)/i);
        let epsStr = '文藝茶館第一季';
        if (epMatch) epsStr += ' EP' + epMatch[1];

        return `  {
    id: '${v.id}',
    title: '${v.title.replace(/'/g, "\\'")}',
    series: 'wenyi_s1',
    category: 'all',
    duration: '${v.duration}',
    description: '${epsStr}'
  },`;
    }).join('\n');

    let newContent = content.replace(
        "export const videos = [",
        "export const videos = [\n" + newVideosStr
    );

    if (!newContent.includes('wenyi_s1')) {
        // Add SERIES
        newContent = newContent.replace(
            "{ id: 'wenyi', label: '文藝茶館 第二季' },",
            "{ id: 'wenyi_s1', label: '文藝茶館 第一季' },\n  { id: 'wenyi', label: '文藝茶館 第二季' },"
        );

        // Add SUBCATEGORIES
        newContent = newContent.replace(
            "export const WENYI_SUBCATEGORIES = [",
            "export const WENYI_S1_SUBCATEGORIES = [\n  { id: 'all', label: '全部單元' }\n];\n\nexport const WENYI_SUBCATEGORIES = ["
        );
    }

    fs.writeFileSync('src/data/videoData.js', newContent);
    console.log('Successfully updated videoData.js');
} catch (e) {
    console.error(e);
}
